### PR TITLE
Phase Table Validation In Muon Analysis

### LINKED ISSUE #30168: Muon phase table
Phase table interaction could be improved.

1. Try to create a phase table without loading data -> unhelpful error
2. First and last good validation
3. Create a phase table and then tick `output fits` and create another table -> errors

### PR DESCRIPTION
**Pull request by Matt-Cumber**
**Description of work.**
Added some validation to first good data and last good data with some unit tests. There was an error message that occurred when trying to create a table whilst outputting fit information. The message occurred as workspace names were being overwritten when grouped together. To avoid this for now I have stopped the grouping for fit information workspace. The fix for calculating with an empty table shouldn't be a problem given the tab is disabled now when no data is loaded which was a fix from #30181 

**To test:**
1. Open Muon Analysis
2. Load a run (e.g. MUSR62260)
3. Go to phase table tab
4. Check calculate table still works as before
5. Now try change first good and last good data
    1. You should not be able to make first > last
    2. first < min or > max
    3. last < min or > max
6. Check output fit information
7. then calculate table. this should not produce an error message as it did before, and the workspace should be in the ADS on it's own
8. Check loading other runs and outputting fit information also works  

Fixes #30168 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py [hunk 1]
```diff
@@ -237,3 +233,23 @@ def update_current_groups_list(self):
...
+
+    def handle_first_good_data_changed(self):
+        self._validate_data_changed(self.view.first_good_time, "First Good Data")
+
+    def handle_last_good_data_changed(self):
+        self._validate_data_changed(self.view.last_good_time, "Last Good Data")
+
+    def _validate_data_changed(self, data, string):
+        first_good_time = self.context.phase_context.options_dict['first_good_time']
+        last_good_time = self.context.phase_context.options_dict['last_good_time']
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py:245]: you probably want to get these values from the main context https://github.com/mantidproject/mantid/blob/master/scripts/Muon/GUI/Common/contexts/muon_context.py#L378 

At the moment it says last cannot be greater than 15, but I know the data is good till 32. You would probably need the largest first and smallest last to make sure they are always valid when multiple runs are loaded.

File: scripts/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py [hunk 2]
```diff
@@ -241,8 +242,11 @@ def handle_last_good_data_changed(self):
         self._validate_data_changed(self.view.last_good_time, "Last Good Data")
 
     def _validate_data_changed(self, data, string):
-        first_good_time = self.context.phase_context.options_dict['first_good_time']
-        last_good_time = self.context.phase_context.options_dict['last_good_time']
+        run = get_run_number_from_workspace_name(self.view.input_workspace,
```
Comment (AnthonyLim23) [inline scripts/Muon/GUI/Common/phase_table_widget/phase_table_presenter.py:245]: do the type conversion here for neatness

**Comment by AnthonyLim23** [comment]
just need something in the release notes and then its good to go

**Comment by Matt-Cumber** [comment]
> just need something in the release notes and then its good to go

Do we need release notes for this since really it is just an addition to #30109?

**Comment by AnthonyLim23** [comment]
The validation of the first and last data is new

**Review by AnthonyLim23 — APPROVED** [review]
looks good. :shipit: 

The latest change is just docs and the last build passed everything.

**Pull request merged**